web/source/keymanweb.js

```diff
@@ -3058,15 +3076,91 @@
 /**
  * Install a keyboard script that has been downloaded from a keyboard server
  * 
- *  @param  {string}  kbdFile   keyboard filename
+ *  @param  {Object}  kbdStub   keyboard filename
  *    
  **/      
-  keymanweb.installKeyboard = function(kbdFile)
+  keymanweb.installKeyboard = function(kbdStub)
   {
     var Lscript = util._CreateElement('SCRIPT');
     Lscript.charset="UTF-8";        // KMEW-89
-    Lscript.src = keymanweb.getKeyboardPath(kbdFile);       
     Lscript.type = 'text/javascript';
+
+    var kbdFile = kbdStub['KF'];
+    var kbdLang = kbdStub['KL'];
+    var kbdName = kbdStub['KN'];
+
+    if(typeof(kbdStub.asyncLoader.callback) == "function") {
```

Comment:
> How much of this code is run when we are using the embedded version?

That's actually the reason we only add `onerror` if we're using asyncLoader.callback.  *That* function relies upon the existence of `util.wait`, which relies upon being 'native' instead of 'embedded.'  I suppose I should document it accordingly, but that check is disabling the control path for embedded systems.

I do this because it appears that KMEA and KeymanWeb interact through the Javascript console, and at present all the `onerror` handler does is print to the console and generate a `util.wait` message, neither of which should occur in embedded systems.